cmake/Find.cmake

```diff
@@ -0,0 +1,39 @@
+# Functions to aid the built-in find_ functions
+
+# Same as find_path, but always search in .deps directory first and then everything else.
+function(find_path2)
```

Comment:
This is a good name imo. The "2" is a strong hint that it's similar to the standard find_path. Using a different name  would reduce intuition while gaining nothing.